packages/elastic-handler/src/elasticHandler.ts

```diff
@@ -1,45 +1,39 @@
 import type {
   LogEntry,
-  LogLevel,
   ReadonlyRecursive,
+  TaskAwareLogHandlerPlugin,
   TContextBase,
   TContextShape,
 } from '@debugr/core';
-import { clone, TaskAwareLogHandler } from '@debugr/core';
+import { LogLevel } from '@debugr/core';
 import { Client } from '@elastic/elasticsearch';
 import { AsyncLocalStorage } from 'async_hooks';
 import { v4 } from 'uuid';
-import { ElasticHandlerOptions, ElasticOptions } from './types';
+import type { ElasticHandlerOptions, ElasticOptions } from './types';
 
 export class ElasticHandler<
   TTaskContext extends TContextBase,
   TGlobalContext extends TContextShape,
-> extends TaskAwareLogHandler<TTaskContext> {
-  public readonly identifier: string = 'elastic';
-
-  public readonly doesNeedFormatters: boolean = false;
-
-  public readonly threshold: LogLevel | number;
+> implements TaskAwareLogHandlerPlugin<TTaskContext, TGlobalContext> {
+  public readonly id: string = 'elastic';
 
   private readonly elasticClient: Client;
 
-  private readonly opts: ElasticHandlerOptions<TTaskContext, TGlobalContext>;
+  private readonly options: ElasticHandlerOptions<TTaskContext, TGlobalContext>;
+
+  private readonly threshold: LogLevel | number;
 
   private readonly asyncStorage: AsyncLocalStorage<string[]>;
 
   private lastError?: Date;
 
-  constructor(opts: ElasticHandlerOptions<TTaskContext, TGlobalContext>, elasticClient: Client) {
-    super();
-    this.threshold = opts.threshold;
-    this.opts = opts;
-    this.elasticClient = elasticClient;
-  }
-
-  public injectPluginManager(): void {}
-
-  public flush(): void {
-    //
+  constructor(options: ElasticOptions<TTaskContext, TGlobalContext>);
```

Comment:
why?

Reply:
Aby typy reflektovaly způsob, kterým ten konstruktor voláš - buď mu předáš nakonfigurovanej `Client`, a pak v `options` nemaj `ClientOptions` z Elasticu co dělat, nebo mu `Client` nepředáš, a pak chceš, aby `options` nejen ty `ClientOptions` _uměly_ dostat, ale aby případně reflektovaly i cokoliv co je v `ClientOptions` required (což teď myslím není nic, ale do budoucna by třeba bejt mohlo). Ve zkratce: je to proto, aby ten constructor byl type-safe.